ansible_collections/arista/avd/plugins/action/eos_designs_facts.py

```diff
@@ -0,0 +1,150 @@
+from ansible_collections.arista.avd.plugins.module_utils.eos_designs_facts import EosDesignsFacts
+from ansible_collections.arista.avd.plugins.module_utils.utils import AristaAvdMissingVariableError
+
+
+class ActionModule(ActionBase):
+    def run(self, tmp=None, task_vars=None):
+        if task_vars is None:
+            task_vars = {}
+
+        result = super().run(tmp, task_vars)
+        del tmp  # tmp no longer has any effect
+
+        cprofile_file = self._task.args.get("cprofile_file")
+        if cprofile_file:
+            profiler = cProfile.Profile()
+            profiler.enable()
+
+        set_avd_switch_facts = self._task.args.get('avd_switch_facts', False)
+
+        groups = task_vars.get('groups', {})
+        fabric_name = task_vars.get('fabric_name', '')
+        fabric_hosts = groups.get(fabric_name, [])
+        ansible_play_hosts_all = task_vars.get('ansible_play_hosts_all', [])
+
+        # Check if fabric_name is set and that all play hosts are part Ansible group set in "fabric_name"
+        if fabric_name is None or not set(ansible_play_hosts_all).issubset(fabric_hosts):
+            raise AnsibleActionFail("Invalid/missing 'fabric_name' variable."
+                                    "All hosts in the play must have the same 'fabric_name' value"
+                                    "which must point to an Ansible Group containing the hosts.")
+
+        # This is not all the hostvars, but just the Ansible Hostvars Manager object where we can retrieve hostvars for each host on-demand.
+        hostvars = task_vars['hostvars']
+
+        # Caveat: Since we load default vars only once, it will be templated based on the vars of the random host triggering this task
+        #         This should not be too bad, since all hosts are within the same fabric - hence they should also use the same "design"
+        default_vars = self._templar.template(self._task._role.get_default_vars())
+        default_vars['ansible_search_path'] = task_vars.get('ansible_search_path')
+
+        if set_avd_switch_facts:
+            avd_overlay_peers = {}
+            avd_topology_peers = {}
+            avd_switch_facts_instances = self.create_avd_switch_facts_instances(fabric_hosts, hostvars, default_vars)
+
+            avd_switch_facts = self.render_avd_switch_facts(avd_switch_facts_instances)
+
+            for host in fabric_hosts:
+                host_evpn_route_servers = avd_switch_facts[host]['switch'].get('evpn_route_servers', [])
+
+                for peer in host_evpn_route_servers:
+                    avd_overlay_peers.setdefault(peer, []).append(host)
+
+                host_topology_peers = avd_switch_facts[host]['switch'].get('uplink_peers', [])
+
+                for peer in host_topology_peers:
+                    avd_topology_peers.setdefault(peer, []).append(host)
+
+            result['ansible_facts'] = {}
+            if set_avd_switch_facts:
```

Comment:
Any reason to check this variable ```set_avd_switch_facts``` as this is already inside ```if set_avd_switch_facts``` (line52)

Reply:
No reason any more. It looked differently earlier. I will clean up.